File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        previous_price = prices[0]
        for idx in range(1, len(prices)):
            curr_price = prices[idx]
            if curr_price > previous_price:
                # sell
                max_profit += (curr_price - previous_price)

                # then "buy" today
                previous_price = curr_price
            else:
                # set the previous_price (i.e. buy price to be curr_price)
                previous_price = curr_price
            
        return max_profit

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var max_profit = 0;
        var buy = prices[0];
        for (int i = 1; i < prices.Length; i++) {
            if (prices[i] <= buy) {
                buy = prices[i];
            }
            else {
                max_profit = Math.Max(max_profit, prices[i] - buy);
            }
        }
        
        return max_profit;        
    }
}
